Enable compiling extensions statically into PHP

jjeising opened the pull request:
This should enable compiling both extensions statically into PHP and fixes #894. I've tested building statically and as a shared module against PHP 7.2.13.

I've tried to introduce only a few extraneous changes. I've removed some duplicate lines from `php_sqlsrv(_int).h` that were already present in `core_sqlsrv.h`. For a short time I also had [this change](https://gist.github.com/jjeising/889e0eb1574cac8d0be6434dc5388b07) included, but is not needed, as the ini definitions will remain in the internal header files. You may be nonetheless interested in cleaning this up a little.

To include both extensions at the same time on Windows there is probably a change needed to `configure_module_dirname + "\\shared"` in `config.w32`, but I'm not familiar with syntax there.

jjeising commented on source/pdo_sqlsrv/config.m4:
```diff
@@ -63,7 +63,10 @@ if test "$PHP_PDO_SQLSRV" != "no"; then
+  if test -f $srcdir/ext/sqlsrv/shared/core_sqlsrv.h; then
+    pdo_sqlsrv_inc_path=$srcdir/ext/sqlsrv/shared/
+    shared_src_class=""
+  elif test -f $srcdir/ext/pdo_sqlsrv/shared/core_sqlsrv.h; then
```
> This change makes it impossible to compile just pdo_sqlsrv without sqlsrv (I get a bunch of undefined reference compilation errors)

Can you elaborate a bit what you tried to do? This should only happen if you copy both extensions to the PHP source `ext/` directory and then specify `--disable-sqlsrv`. As this should be required rarely I thought it might be better than no support. But it is probably possible to test if `sqlsrv` is enabled, [like this](https://github.com/Microsoft/msphpsql/blob/master/source/pdo_sqlsrv/config.m4#L25). I'll have a look.

jjeising replied:
@david-puglielli I've added the additional check. Had to swap between the extensions, because `pdo_sqlsrv` is included first by `buildconf`. This might be a bit fragile, but enables me to build both without arguments and with `--with-pdo_sqlsrv --disable-sqlsrv` and also with both extensions active. Does this address your concerns?

david-puglielli replied:
Ok, I think your recent changes address my concerns. I've tested every combination, and the only one that doesn't work is pdo_sqlsrv shared and sqlsrv static, but I don't think that's a big problem. We will continue testing and see if we can get it working on Windows as well.

david-puglielli commented:
We will be merging this PR as soon as we figure out how to implement this on Windows as well, but we can't before then. Please feel free to update this PR if you have any suggestions.

jjeising commented:
Thanks for your additional work! 🎉

> Please see my changes at the static-compilation branch of david-puglielli/msphpsql and incorporate them in this PR.

I've squashed the missing extern declaration and updated `config.w32`. I've tried to reduce duplicate code there.
